xcape/core.py

```diff
@@ -129,40 +129,78 @@ def calc_cape(p, t, td, ps, ts, tds, **kwargs):
     zMUlev : array-like
         height of MUlev (m)
      """
-    
+    if len(args)>1:
+        print('too many optional arguments')
+        
+    elif kwargs['vertical_lev']!=True:
+        print('please select type of vertical grid, pressure level, or model level')
+        
+    elif (len(args)==0)&(kwargs['vertical_lev'] == 'pressure'):
+        print('vertical grid is set to pressure levels,\n'+
+                +'but location in fortran values (1: nlev) of where p <= ps is missing')
+        
+    elif (len(args)==1)&(args[0].shape!=ps.shape)&(kwargs['vertical_lev'] == 'pressure'):
+        print('vertical grid is set to pressure levels,\n'+
+                +'but location in fortran values (1: nlev) of where p <= ps is not the correct shape\n'+
+                +'(pres_lev_pos.shape should be equal to ps.shape)')
+        
```

Comment:
This is the part that would need some `break` in it I guess. the idea is to do all the checks here, so that then in the following steps I don't have to worry about it.

Reply:
raise don't print